### ISSUE TITLE
Clang believes `template<...> int foo() requires ...` is more constrained than `template<...> requires ... int foo()`

### ISSUE THREAD

**Issue by Quuxplusone** [saarraz/clang-concepts-old#44]
https://concepts.godbolt.org/z/5TrumP

This test case passes, with Clang, when I would naturally expect it to be ill-formed:
```
template<class T>
concept Foo = true;

template<class T>
constexpr int foo(T)  // A
   requires Foo<T>
{
    return 1;
}

template<class T>
  requires Foo<T>
constexpr int foo(T)  // B
{
    return 2;
}

static_assert(foo(0) == 1);
```

Notice the *different position of the requires-clause* in `B`.

GCC considers template `B` to be an invalid redefinition of template `A`. Clang considers `A` and `B` to be two different templates.
Here's a test case showing that GCC believes there's only one template, while Clang believes there are two templates: https://concepts.godbolt.org/z/WrMQeW
I think Clang's behavior makes more sense than GCC's so far — `A` and `B` *should* be different templates, as I understand the working draft. However, here's where it gets weird.

Clang apparently believes that `A` is *a better match* than `B`!  (https://wg21.link/over.match.best) So `foo(0)` is not ambiguous; it resolves to `A`.

I am fairly confident that this can't be right. The two templates have exactly the same associated constraints, and so neither one should be "more constrained."

**Comment by saarraz** [comment]
http://eel.is/c++draft/temp.over.link#7 does indeed indicate that these are not redeclarations, so the bug here is indeed the subsumption

**Comment by saarraz** [comment]
Fixed in 9352f531bc29fa8ba4a4640797632aa0cc73a854
